client/components/plans/plans-compare/index.jsx

```diff
@@ -153,17 +154,17 @@ var PlansCompare = React.createClass( {
 	comparisonTable: function() {
 		var plansColumns,
 			featuresList = this.props.features.get(),
+			hideFreePlan = this.props.hideFreePlan ? true : false,
+			numberOfPlaceholders = 4,
 			plans = this.props.plans.get(),
-			site = this.props.selectedSite,
-			showJetpackPlans = site ? site.jetpack : false;
+			site = this.props.selectedSite;
 
 		plans = plans.filter( function( plan ) {
-			return ( showJetpackPlans === ( 'jetpack' === plan.product_type ) );
+			return filterPlansBySiteAndProps( plan, site, hideFreePlan );
```

Comment:
Also, I wouldn't group 2 differents type of filtering together.
Why not just have:
```js
if ( site ) {
    plans = plans.filter( function( plan ) {
         return site.jetpack ? isJetpackPlan( plan ) : ! isJetpackPlan( plan );
    } );
}

if ( hideFreePlan ) {
    plans = plans.filter( function( plan ) {
         return ! isFreePlan( plan );
    } );
}
```
It may be a bit longer here but I think it's self explanatory.